### PR TITLE
[feat] Zone Analytics

### LINKED ISSUE #41: Implement Zone Analytics endpoints
https://api.cloudflare.com/#zone-analytics

### PR DESCRIPTION
**Pull request by nishanths**
Resolves #41

### CODE REVIEW

**Comment by elithrar** [comment]
Are you also able to submit tests for this? Refer to https://github.com/cloudflare/cloudflare-go/blob/master/user_test.go for guidance.

**Comment by nishanths** [comment]
@elithrar done

File: zone.go [hunk 1]
```diff
@@ -107,6 +108,71 @@ type ZoneSettingResponse struct {
...
 }
 
+// ZoneAnalyticsData contains totals and timeseries analytics data for a zone.
+type ZoneAnalyticsData struct {
+	Totals     ZoneAnalytics   `json:"totals"`
+	Timeseries []ZoneAnalytics `json:"timeseries"`
+}
+
+// ZoneAnalyticsDataResponse represents the response from the Zone Analytics Dashboard endpoint.
+type ZoneAnalyticsDataResponse struct {
```
Comment (elithrar) [inline zone.go]: Ref #78 - make this private.

File: zone.go [hunk 2]
```diff
@@ -107,6 +108,71 @@ type ZoneSettingResponse struct {
...
+}
+
+// ZoneAnalyticsColocation contains analytics data by datacenter.
+type ZoneAnalyticsColocation struct {
+	ColocationID string          `json:"colo_id"`
+	Timeseries   []ZoneAnalytics `json:"timeseries"`
+}
+
+// ZoneAnalyticsColocationResponse represents the response from the Zone Analytics By Co-location endpoint.
+type ZoneAnalyticsColocationResponse struct {
```
Comment (elithrar) [inline zone.go]: Ref #78 - make this private.

**Comment by elithrar** [comment]
Just to be a huge pain: a couple more comments related to #78. Otherwise: LGTM!

**Comment by nishanths** [comment]
@elithrar done 👍

**Comment by elithrar** [comment]
LGTM. @jamesog @Tenzer ?

**Comment by Tenzer** [comment]
As long as we open a new issue stating that we need to add support for the extra parameters, it's good to me.

**Comment by elithrar** [comment]
In order to avoid breaking the API more than we need to, we should
implement the optional params before merging.

No rush on that work at all however (enjoy your long weekends!)

**Comment by nishanths** [comment]
@elithrar @Tenzer Thoughts?

**Comment by jamesog** [comment]
I think that looks great. In it goes :-)

**Pull request merged**